### DISCUSSION TITLE
Location Detection - Can we get rid of GPS?

### DISCUSSION THREAD

**Discussion by Derek-Gutheil** [Ribbit-Network/ribbit-network-frog-hardware#111] [Ideas]
GPS might be overkill, especially since we are purposefully obfuscating the frog's location.

Can we use other methods like an IP based one such as:
https://www.geoplugin.com/
https://ip-api.com/
https://ipstack.com/
etc.

I just ran Geoplugin myself and it claimed an accuracy radius of 5 miles.

Or even a nearby WiFi network technique that uses google maps (may cost money):
https://developers.google.com/maps/documentation/geolocation/overview

Thoughts?

**Comment by keenanjohnson** [comment]
It's a great question @Derek-Gutheil! I think someone asked me something similar in person once, and this conversation is making me think that perhaps we need some sort of engineering knowledge wiki or something to capture decisions already explored.

Anyway, we would like to achieve an accuracy of around +/- 1 km to achieve the best science we can while trying to obfuscate the exact location of someone. [See this discussion if you are interested!](https://github.com/Ribbit-Network/ribbit-network-dashboard/discussions/27#discussioncomment-1291523) Geoplugin or other IP address related position technologies just can't achieve anything close to that. Balena uses a IP based location technology for location and the accuracy in practice that I've seen can be vastly worse than 5 miles. At my home for instance, it's about 25 miles off.

I haven't had experience with the nearby wifi technique, but it seems like that approach could be quite unpredictable? I imagine it might work well in a large city, but would be perhaps useless in a rural setting.

**Reply by Derek-Gutheil** [reply]
Awesome, I checked out that other thread and have a much better undestanding. I am now wondering if we have any sort of setup process that the user undergoes when they turn the frog on. If so, we could welcome the user, have them put in WiFi information, and then have them share their phones location with the Frog.

**Reply by keenanjohnson** [reply]
We only have a setup process to connect to Wifi, but certainly such a different setup process could be made. We'll just need to consider the edge cases. What if they move the sensor etc and if the cost savings from lack of GPS actually benefit the user :)
